uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/installer/PlayStore.java

```diff
@@ -145,36 +129,43 @@ private void installOrUpdateAppFromMarketPage(String playStoreAction) {
         Logger.i(TAG, "Performing " + playStoreAction + " App From Market Page Internal..");
-        final UiDevice device = UiDevice.getInstance(getInstrumentation());
-        final UiObject uiObjBtn = device.findObject(
-                new UiSelector().className(Button.class).text(playStoreAction).enabled(true)
-        );
-
-        uiObjBtn.waitForExists(FIND_UI_ELEMENT_TIMEOUT);
-        uiObjBtn.click();
 
-        checkInstallCompletion();
+        // There is some inconsistency in how the PlayStore buttons are appearing, regardless of API level
+        try {
+            final UiObject uiObjBtn = UiAutomatorUtils.obtainUiObjectWithUiSelector(
+                    new UiSelector().description(playStoreAction),
+                    PLAY_STORE_UI_OBJECT_TIMEOUT);
+            uiObjBtn.click();
+            checkInstallCompletion(new UiSelector().description("Uninstall"));
+        } catch (final UiObjectNotFoundException e) {
+            Logger.i(TAG, "Got exception when trying to find Install/Update button for PlayStore: " + e.getMessage());
+            final UiObject uiObjBtn = UiAutomatorUtils.obtainUiObjectWithUiSelector(
+                    new UiSelector().className(Button.class).text(playStoreAction).enabled(true),
+                    PLAY_STORE_UI_OBJECT_TIMEOUT);
+            uiObjBtn.click();
+            checkInstallCompletion(new UiSelector().className(Button.class).text("Uninstall").enabled(true));
+        }
     }
 
-    private void checkInstallCompletion() {
-        final UiDevice device = UiDevice.getInstance(getInstrumentation());
-        final UiObject openButton = device.findObject(
-                new UiSelector().className(Button.class).text("Open").enabled(true)
-        );
-
-        openButton.waitForExists(PLAY_STORE_INSTALL_OR_UPDATE_APP_TIMEOUT);
+    private void checkInstallCompletion(@NonNull final UiSelector selector) {
+        final UiObject openButton = UiAutomatorUtils.obtainUiObjectWithUiSelector(selector, PLAY_STORE_INSTALL_OR_UPDATE_APP_TIMEOUT);
+        Assert.assertTrue(openButton.exists());
     }
 
     private void acceptGooglePlayTermsOfService() {
-        Logger.i(TAG, "Accept Google Play Terms Of Service while installing App from Playstore..");
-        final UiObject termsOfService = UiAutomatorUtils.obtainUiObjectWithText("Terms of Service");
-        Assert.assertTrue(termsOfService.exists());
-        final UiObject acceptBtn = UiAutomatorUtils.obtainUiObjectWithText("ACCEPT");
-        Assert.assertTrue(acceptBtn.exists());
-        try {
-            acceptBtn.click();
-        } catch (UiObjectNotFoundException e) {
-            throw new AssertionError(e);
+        if (shouldHandleTermsAndServices) {
+            Logger.i(TAG, "Accept Google Play Terms Of Service while installing App from Playstore..");
+            final UiObject termsOfService = UiAutomatorUtils.obtainUiObjectWithText("Terms of Service");
```

Comment:
Can we define the strings used as constants at the top? I think it will be helpful to understand what text elements we are looking for quickly.

Reply:
I feel like it would make more sense to do this as part of a larger refactoring effort for UI library. I'm inclined to keep this PR at this size and tackle that work at a later time.